codalab/codalab-worksheets
# Allow async loading of file contents

## Description
fixes #2244 

When `brief=1` is passed to the `/rest/interpret` endpoint, file contents are not resolved (i.e., `interpret_genpath_table_contents` is not called).

Then, unresolved columns look like this:

![image](https://user-images.githubusercontent.com/1689183/81232772-760a4280-8fc3-11ea-823f-b8c7460503ac.png)

Async requests are then sent to `/rest/interpret/genpath-table-contents`, and the table contents are then resolved.

## Testing

1. Upload a folder with a file called `package.json` with the contents `{"version": 2}`
1. Add the following in the worksheet (replace 0xc30ee021930f4bb3a9671753d2a92094 with the bundle id)

```
% schema cub
% add uuid uuid '[0:8]'
% add name
% add output_value /package.json:version
% add time time duration
% add state
% add description

% display table cub
[dataset bundle]{0xc30ee021930f4bb3a9671753d2a92094}
[dataset bundle]{0xc30ee021930f4bb3a9671753d2a92094}
[dataset bundle]{0xc30ee021930f4bb3a9671753d2a92094}
[dataset bundle]{0xc30ee021930f4bb3a9671753d2a92094}
[dataset bundle]{0xc30ee021930f4bb3a9671753d2a92094}
[dataset bundle]{0xc30ee021930f4bb3a9671753d2a92094}
[dataset bundle]{0xc30ee021930f4bb3a9671753d2a92094}
[dataset bundle]{0xc30ee021930f4bb3a9671753d2a92094}
[dataset bundle]{0xc30ee021930f4bb3a9671753d2a92094}
[dataset bundle]{0xc30ee021930f4bb3a9671753d2a92094}
```

1. The worksheet should now async load file contents.
